pkg/provider/onboardbase/client.go

```diff
@@ -0,0 +1,224 @@
+			return fmt.Errorf(errInvalidClusterStoreMissingOnboardbaseAPIKeyNamespace)
+		}
+		objectKey.Namespace = *c.store.Auth.OnboardbaseAPIKey.Namespace
+	}
+
+	err := c.kube.Get(ctx, objectKey, credentialsSecret)
+	if err != nil {
+		return fmt.Errorf(errFetchOnboardbaseAPIKeySecret, err)
+	}
+
+	onboardbaseAPIKey := credentialsSecret.Data[c.store.Auth.OnboardbaseAPIKey.Key]
+	if (onboardbaseAPIKey == nil) || (len(onboardbaseAPIKey) == 0) {
+		return fmt.Errorf(errMissingOnboardbaseAPIKey, c.store.Auth.OnboardbaseAPIKey.Key, credentialsSecretName)
+	}
+	c.onboardbaseAPIKey = string(onboardbaseAPIKey)
+
+	onboardbasePasscode := credentialsSecret.Data[c.store.Auth.OnboardbasePasscode.Key]
+	if (onboardbasePasscode == nil) || (len(onboardbasePasscode) == 0) {
+		return fmt.Errorf(errMissingOnboardbasePasscode, c.store.Auth.OnboardbasePasscode.Key, credentialsSecretName)
+	}
+
+	c.onboardbasePasscode = string(onboardbasePasscode)
+
+	return nil
+}
+
+func (c *Client) Validate() (esv1beta1.ValidationResult, error) {
+	timeout := 15 * time.Second
+	clientURL := c.onboardbase.BaseURL().String()
+
+	if err := utils.NetworkValidate(clientURL, timeout); err != nil {
+		return esv1beta1.ValidationResultError, err
+	}
+
+	if err := c.onboardbase.Authenticate(); err != nil {
+		return esv1beta1.ValidationResultError, err
+	}
+
+	return esv1beta1.ValidationResultReady, nil
+}
+
+func (c *Client) DeleteSecret(ctx context.Context, remoteRef esv1beta1.PushRemoteRef) error {
+	return fmt.Errorf("not implemented")
+}
+
+func (c *Client) PushSecret(ctx context.Context, value []byte, remoteRef esv1beta1.PushRemoteRef) error {
+	return fmt.Errorf("not implemented")
+}
+
+func (c *Client) GetSecret(_ context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) ([]byte, error) {
+	request := dClient.SecretRequest{
+		Project:     c.project,
+		Environment: c.environment,
+		Name:        ref.Key,
+	}
+
+	secret, err := c.onboardbase.GetSecret(request)
+	if err != nil {
+		return nil, fmt.Errorf(errGetSecret, ref.Key, err)
+	}
+
+	return []byte(secret.Value), nil
+}
+
+func (c *Client) GetSecretMap(ctx context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) (map[string][]byte, error) {
+	data, err := c.GetSecret(ctx, ref)
+	if err != nil {
+		return nil, err
+	}
+
+	kv := make(map[string]json.RawMessage)
+	err = json.Unmarshal(data, &kv)
+	if err != nil {
+		return nil, fmt.Errorf(errUnmarshalSecretMap, ref.Key, err)
+	}
+
+	secretData := make(map[string][]byte)
+	for k, v := range kv {
+		var strVal string
+		err = json.Unmarshal(v, &strVal)
+		if err == nil {
+			secretData[k] = []byte(strVal)
+		} else {
+			secretData[k] = v
+		}
+	}
+	return secretData, nil
+}
+
+func (c *Client) GetAllSecrets(ctx context.Context, ref esv1beta1.ExternalSecretFind) (map[string][]byte, error) {
+	secrets, err := c.getSecrets(ctx)
+	selected := map[string][]byte{}
+
+	if err != nil {
+		return nil, err
+	}
+
+	if ref.Name == nil && ref.Path == nil {
+		return secrets, nil
+	}
```

Comment:
If users deploy `dataFrom.find.tags` like the example below, the provider should error out as this feature is not supported by the provider: 

```
  dataFrom:
  - find: # this should error out - right now it is returning all secrets
      tags:
        environment: "prod"
        application: "app-name"
```

Reply:
> hey @limistah ! It looks better now. I added some use cases that are not yet covered and might need some tweaks (and tests).
> 
> there are also some linting errors happening. You can reproduce them by typing `make lint`. To also check all of the files that we generate, you can run `make reviewable`.

Thank you @gusfcarvalho 

I'd take another good look.